File: 1291-Sequential Digits.py
class Solution(object):
    def sequentialDigits(self, low, high):
        """
        :type low: int
        :type high: int
        :rtype: List[int]
        """
        candidate = "123456789"
        N = 9
        res = []
        
        for length in range(len(str(low)), len(str(high))+1):
            for start in range(0, N-length+1):
                currNum = int(candidate[start:start+length])
                if low<=currNum<=high:
                    res.append(currNum)
                elif currNum > high:
                    break
        
        return res

File: C++/1291-Sequential Digits.cpp
class Solution {
public:
    vector<int> sequentialDigits(int low, int high) {
        string str = "123456789";
        vector<int> nums;
        for (int length = to_string(low).size(); length <= to_string(high).size(); length++){
            for (int start=0; start < 10-length; start++) {
                int num = stoi(str.substr(start, length));
                if (num >= low && num <= high) nums.push_back(num);
            }
        }
        return nums;
    }
};